optimizely/odp/odp_manager.py

```diff
@@ -42,20 +42,27 @@ def __init__(
         self.odp_config = OdpConfig()
         self.logger = logger or optimizely_logger.NoOpLogger()
 
-        self.segment_manager = None
-        self.event_manager = None
+        self.segment_manager = segment_manager
+        self.event_manager = event_manager
 
         if not self.enabled:
             self.logger.info('ODP is disabled.')
             return
 
-        if segment_manager:
-            segment_manager.odp_config = self.odp_config
-            self.segment_manager = segment_manager
-        else:
-            self.segment_manager = OdpSegmentManager(self.odp_config,
-                                                     segments_cache,
-                                                     ZaiusGraphQLApiManager(logger), logger)
+        if not self.segment_manager:
+            if not segments_cache:
+                segments_cache = LRUCache(
+                    OdpSegmentsCacheConfig.DEFAULT_CAPACITY,
+                    OdpSegmentsCacheConfig.DEFAULT_TIMEOUT_SECS
+                )
+            self.segment_manager = OdpSegmentManager(
+                self.odp_config,
+                segments_cache,
+                ZaiusGraphQLApiManager(logger), logger
+            )
+
+        self.segment_manager.odp_config = self.odp_config
```

Comment:
```suggestion
        else:
              self.segment_manager.odp_config = self.odp_config
```
Logically same, just for clarification

Reply:
Thx, good one. Fixed

Reply:
This also avoids odp_config being called twice